pkg/resourceapply/apps_test.go

```diff
@@ -493,6 +497,60 @@ func TestApplyStatefulSet(t *testing.T) {
 			expectedErr:     nil,
 			expectedEvents:  []string{"Normal StatefulSetUpdated StatefulSet default/test updated"},
 		},
+		{
+			name: "deletes and creates the StatefulSet when selector differs",
```

Comment:
when selector is changed and still matches the old pods

Reply:
done